services/should-show-out-of-france-banner.js

```diff
@@ -1,14 +1,43 @@
 import { config } from '~/config/environment'
 
 export async function shouldShowOutOfFranceBanner(baseUrl) {
-  const FR_COUNTRY_CODE = 'FR'
-  const url = new URL('/geolocate', baseUrl)
-
   if (!config.isFrenchDomain || !config.isPixSite) return false
 
+  const country = await _getUserCountryFromGeolocationService(baseUrl)
+  if (!country) return false
+
+  return _shouldShowBannerForCountry(country)
+}
+
+async function _getUserCountryFromGeolocationService(baseUrl) {
+  const url = new URL('/geolocate', baseUrl)
+
   const response = await fetch(url).then((res) => res.json())
+  if (!response) return undefined
 
-  if (!response?.country) return false
+  return response.country
+}
+
+function _shouldShowBannerForCountry(country) {
+  const FR_COUNTRY_CODE = 'FR'
```

Comment:
Compte tenu que le code de Pix en général manipule à la fois des langues, des locales et des pays, `FR_COUNTRY_CODE` gagnerait sûrement à être renommée en `FRANCE_COUNTRY_CODE` pour être explicite sur le fait qu'il s'agit du territoire français de métropole, et non de la langue française.

```suggestion
  const FRANCE_COUNTRY_CODE = 'FR'
```